### PR TITLE
Diagnostic Session Control & UI Changes

### PR DESCRIPTION
**Pull request by asavu91**
## Description

Added diagnostic session control endpoint. Small changes to UI.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/rest_api/actions/base_actions.py [hunk 1]
```diff
@@ -193,11 +193,11 @@ def __collect_response(self, sid: int):
         return None
 
     def __verify_frame(self, msg: can.Message, sid: int):
+
         log_info_message(logger, f"Verifying frame with SID: {sid:02X}, message data: {[hex(byte) for byte in msg.data]}")
 
-        # Existing verification logic...
-        if msg.arbitration_id % 0x100 != self.my_id:
-            return False
+        if msg.arbitration_id == self.my_id or msg.arbitration_id == (self.my_id >> 8) | ((self.my_id & 0xFF) << 8):
```
Comment (CatalinGP) [inline src/rest_api/actions/base_actions.py:199]: does check if API ID is properly send in HEX?

File: src/rest_api/actions/base_actions.py [hunk 2]
```diff
@@ -340,6 +340,30 @@ def _authentication(self, id):
...
+
+        Args:
+        - id: The identifier of the target ECU or device.
+        - sub_funct: The sub-function code to change the session.
+
+        Returns:
+        - A dictionary with the status and message of the operation.
+        """
+        try:
+            session_type = "DEFAULT" if sub_funct == 0x01 else "PROGRAMMING" if sub_funct == 0x02 else "unknown"
```
Comment (CatalinGP) [inline src/rest_api/actions/base_actions.py:355]: session type would help if passed as argument?

File: src/rest_api/actions/base_actions.py [hunk 3]
```diff
@@ -340,6 +340,30 @@ def _authentication(self, id):
...
+            log_info_message(logger, f"Changing session to {session_type} (sub_function: {sub_funct})")
+
+            self.generate.session_control(id, sub_funct)
+            frame_response = self._passive_response(SESSION_CONTROL, "Error changing session control")
+
+            if frame_response.data[1] == 0x50:
+                log_info_message(logger, f"Session changed to {session_type} successfully")
+                return {"status": "success", "message": "Session changed successfully"}
+        except CustomError as e:
+            return {"status": "error", "message": str(e)}
```
Comment (CatalinGP) [inline src/rest_api/actions/base_actions.py:365]: does it yield to the json response?

**Review by CatalinGP — COMMENTED** [review]
see my comments:-)

File: src/rest_api/actions/base_actions.py (hunk 2 above)
Comment (asavu91) [reply src/rest_api/actions/base_actions.py:355]: It describes the type of session can be useful for logging, debugging, and making your code more readable.

File: src/rest_api/actions/base_actions.py (hunk 3 above)
Comment (asavu91) [reply src/rest_api/actions/base_actions.py:365]: Yes

**Review by Elisei-26 — APPROVED** [review]

**Review by CatalinGP — APPROVED** [review]

**Pull request merged**